core/src/main/java/be/orbinson/aem/dictionarytranslator/services/DictionaryService.java

```diff
@@ -33,11 +46,52 @@ public interface DictionaryService {
 
     boolean keyExists(Resource dictionaryResource, String language, String key);
 
+    /**
+     * 
+     * @param languageResource
+     * @param key
+     * @return
+     * @deprecated use {@link #getMessages(Resource, String)} instead
+     */
+    @Deprecated
     Resource getMessageEntryResource(Resource languageResource, String key);
 
+    @Deprecated
+    /**
+     * 
+     * @param resourceResolver
+     * @param dictionaryResource
+     * @param language
+     * @param key
+     * @param message
+     * @throws PersistenceException
+     * @deprecated Use {@link #updateMessageEntry(ResourceResolver, Resource, String, String, String)} instead
```

Comment:
`createMessageEntry` and `updateMessageEntry` have a big overlap, imho just `updateMessageEntry` is enough which creates if not yet there.